### DISCUSSION TITLE
Help with discord webhook

### DISCUSSION THREAD

**Discussion by smosse** [lloesche/valheim-server-docker#431] [Q&A]
Hey ! I am sure I am doing something wrong.
I want to send message to my discord. 
The PRE_BOOTSTRAP_HOOK is working but not the other.
What am I doing wrong ?
Here's my docker env

SERVER_NAME=xxxx
WORLD_NAME=xxxxx
SERVER_PASS=xxxxx
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
DEBIAN_FRONTEND=noninteractive
SUPERVISOR_HTTP=true
SUPERVISOR_HTTP_PASS=xxxxx
DISCORD_WEBHOOK=https://discord.com/api/webhooks/927555691919585………………
DISCORD_MESSAGE=Starting Valheim server 
PRE_BOOTSTRAP_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$DISCORD_MESSAGE\"}" "$DISCORD_WEBHOOK"
DISCORD_MESSAGE-RESTART="Restarting Valheim server in one minute!"
PRE_RESTART_HOOK='curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$DISCORD_MESSAGE-RESTART\"}" "$DISCORD_WEBHOOK" && sleep 60’
DISCORD_MESSAGE-PRESHUTDOWN=« Shutdown Valheim server in one minute!"
PRE_SERVER_SHUTDOWN_HOOK='curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$DISCORD_MESSAGE-PRESHUTDOWN\ »} » "$DISCORD_WEBHOOK"
DISCORD_MESSAGE-SHUTDOWN=« Valheim server is Down!"
POST_SERVER_SHUTDOWN_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$DISCORD_MESSAGE-SHURDOWN)\ »} » "$DISCORD_WEBHOOK"

**Comment by tomcoolpxl** [comment]
Here's the **valheim.env** file that works for me:

```
SERVER_NAME=xxxx
WORLD_NAME=xxxxx
SERVER_PASS=xxxxx
SERVER_PUBLIC=true
STATUS_HTTP=true
VALHEIM_PLUS=true
SUPERVISOR_HTTP=true
TZ=Europe/Brussels
DISCORD_WEBHOOK=https://discord.com/api/webhooks/927555......
PRE_BOOTSTRAP_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"[$(date +'%b %d %T')] $DISCORD_MESSAGE\"}" "$DISCORD_WEBHOOK"
PRE_SERVER_SHUTDOWN_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"[$(date +'%b %d %T')] Stopping Valheim Server\"}" "$DISCORD_WEBHOOK"
POST_SERVER_LISTENING_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"[$(date +'%b %d %T')] Valheim Server ready.\"}" "$DISCORD_WEBHOOK"
POST_RESTART_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"[$(date +'%b %d %T')] Valheim Server restarted by Updater.\"}" "$DISCORD_WEBHOOK"
VALHEIM_LOG_FILTER_CONTAINS_Spawned=Got character ZDOID from
ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned={ read l; l=${l:46}; l=${l// :*/}; msg="[$(date +'%b %d %T')] Player $l connected."; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }
```
